crates/ra_hir_def/src/adt.rs

```diff
@@ -58,26 +62,75 @@ pub struct FieldData {
     pub visibility: RawVisibility,
 }
 
+#[derive(Debug, Clone, PartialEq, Eq)]
+pub enum ReprKind {
+    Packed,
+    Other,
+}
+
+fn repr_from_value(item_tree: &ItemTree, of: AttrOwner) -> Option<ReprKind> {
+    item_tree.attrs(of).iter().find_map(|a| {
+        if let Attr {
+            path: ModPath { kind: PathKind::Plain, segments },
+            input: Some(AttrInput::TokenTree(subtree)),
+        } = a
+        {
+            if segments.len() == 1 && segments[0].to_string() == "repr" {
+                parse_repr_tt(subtree)
+            } else {
+                None
+            }
+        } else {
+            None
+        }
+    })
```

Comment:
This is kind of hard to read... Seems like this could be simplified, by some combination of:

* filter_map for pre-filtering of atttibutes
* introducing a helper on `Attrs` to lookup value by key (might be a good idea to check other usages of attrs)
* adding a known name for `"repr"`

Reply:
There was an existing `by_key` method that existed on `Attrs` that I hadn't noticed. I used that and the code is much simpler. It takes `&'static str` though, so I couldn't take advantage of adding a known name for `"repr"`.